### PR TITLE
HUB-18210 Setup database to allow client certificate authentication

### PR DESCRIPTION
**Pull request by romeara**
Enable features such as foreign data wrappers between databases within
the container by generating and storing client certificates for
authentication

### CODE REVIEW

**Comment by romeara** [comment]
Note: I do not have authorization to merge this. Please either grant permission or merge once approved

File: hub-database.sh [hunk 1]
```diff
@@ -87,12 +102,87 @@ else
...
+then
+
+  # Convert naming to agnostic standard
+  if [ -f /tmp/blackduck_user.crt ]
+  then
+  	echo "Renaming client certificate"
+  	
+  	# Convert client key to usable format
+    mv /tmp/blackduck_user.crt /tmp/hub-db-user.crt
+    rm -f /tmp/blackduck_user.crt
```
Comment (paykin) [inline hub-database.sh]: Why `rm` if you do `mv`?

**Comment by paykin** [comment]
What is "data wrappers between databases within the container"?

File: hub-database.sh (hunk 1 above)
Comment (romeara) [reply hub-database.sh]: To keep the file state consistent - I don't want to leave errant files around. Unless... wait, mv is effectively rename, isn't it (facepalm)

**Comment by romeara** [comment]
@paykin Foreign data wrappers - allows setup of one database to read data from another. Specifically, we are going to use this to make a much more efficient reporting database process

File: hub-database.sh (hunk 1 above)
Comment (romeara) [reply hub-database.sh]: Extra "rm"'s removed

File: hub-database.sh [hunk 2]
```diff
@@ -31,8 +31,9 @@ echo "Database host: $targetDatabaseHost"
 ## Until we need another requirement from customers that they need to make a change on pg_hba.conf
 ## trigger this each time when starting a container.
 ## (needs to be kept in sync with 2-hub-setup.sh)
-if [ -d "$PGDATA" ] && [ -s "$PGDATA/PG_VERSION" ];
+if [ -d "$PGDATA" ] && [ -s "$PGDATA/$PG_VERSION" ];
```
Comment (blackduck-cookt) [inline hub-database.sh]: This change is not correct.  PG_VERSION is a file that is created in PGDATA by initdb.  This test is checking that the data directory exists and that initdb has been run.

File: hub-database.sh [hunk 3]
```diff
@@ -48,14 +49,26 @@ hostnossl all             	all                   0.0.0.0/0                 rejec
 EOF
 fi
 
-## Certificate management
+dataPopulated=false
+
+if [ -d "$PGDATA" ] && [ -s "$PGDATA/$PG_VERSION" ];
```
Comment (blackduck-cookt) [inline hub-database.sh]: PG_VERSION, not $PG_VERSION

File: hub-database.sh [hunk 4]
```diff
@@ -48,14 +49,26 @@ hostnossl all             	all                   0.0.0.0/0                 rejec
...
+then
+	echo "Data directory not populated"
```
Comment (blackduck-cookt) [inline hub-database.sh]: I think the "not" is in the wrong branch.

File: hub-database.sh (hunk 2 above)
Comment (romeara) [reply hub-database.sh]: Change reverted, should be using "PG_VERSION" in all instances now

File: hub-database.sh (hunk 3 above)
Comment (romeara) [reply hub-database.sh]: Corrected

File: hub-database.sh (hunk 4 above)
Comment (romeara) [reply hub-database.sh]: Corrected

**Review by paykin — APPROVED** [review]

**Review by blackduck-cookt — APPROVED** [review]

File: Dockerfile [hunk 5]
```diff
@@ -1,4 +1,4 @@
-FROM blackducksoftware/hub-docker-common:1.0.1 as docker-common
+FROM blackducksoftware/hub-docker-common:1.0.3 as docker-common
```
Comment (bd-skatzman) [inline Dockerfile:1]: I am assuming a subsequent PR on hub-backend will bump the versions?

**Review by bd-skatzman — APPROVED** [review]

File: Dockerfile (hunk 5 above)
Comment (romeara) [reply Dockerfile:1]: Of this container, yes

**Pull request merged**